src/circuit/circuit_frame.rs

```diff
@@ -3118,8 +3134,17 @@ fn reduce_cons<F: LurkField, CS: ConstraintSystem<F>, C: Coprocessor<F>>(
                         F::from(arity as u64),
                     )?;
 
-                    let (result_expr, result_env, result_cont) =
-                        coproc.synthesize(cs, g, store, &inputs[..arity], env, cont)?;
+                    let dummy_or_blank = head_zptr != Some(*z_ptr);
```

Comment:
I'm a bit uncomfortable by the fact that although this is incidentally true, it doesn't feel like the correct plumbing

Reply:
We know that the mismatch between `head_zptr` and `Some(*z_ptr)` will happen because it would be too much of a coincidence for the match to happen when we feed the circuit with dummy data. In this sense, it is correct. But I think we should know this info by a different source, explicitly coded.

Reply:
I understand what you're getting at now. I've pushed a commit that also takes into account the 'outer' `not_dummy` value applicable to the whole `reduce_cons` circuit. That will deal with the case you mean — that some completely other cons might have been slot-allocated and match on this call to a coprocessor.

For clarity, the dummy case this code was already actually concerned with is the one in which that outer `not_dummy` is *true* — but the particular coprocessor being synthesized in this loop iteration doesn't match. In that case, the coprocessor circuit still needs to synthesize dummy witness values.